tcs-persistence/src/main/java/com/transformuk/hee/tis/tcs/service/model/Post.java

```diff
@@ -86,7 +87,10 @@
   @OneToMany(mappedBy = "post", fetch = FetchType.LAZY)
   private Set<Placement> placementHistory = new HashSet<>();
   @OneToMany(mappedBy = "post", fetch = FetchType.LAZY)
+  @JsonManagedReference
```

Comment:
This was not actually related to the fetch type for Posts (LAZY or EAGER) but something I noticed was causing some infinite recursion errors in JSON serialization, e.g.
```
Could not write JSON: Infinite recursion (StackOverflowError); nested exception is com.fasterxml.jackson.databind.JsonMappingException: Infinite recursion (StackOverflowError) (through reference chain: com.transformuk.hee.tis.tcs.service.model.Post["associatedTrusts"]->
org.hibernate.collection.internal.PersistentSet[0]->
com.transformuk.hee.tis.tcs.service.model.PostTrust["post"]->
com.transformuk.hee.tis.tcs.service.model.Post["associatedTrusts"]->
```
This may require review and/or more testing to be added

Reply:
May I ask how I can replicate this issue, please?
There're many 1 to many mapping properties in the Post class, so just a little wondering if this happens on others.

Reply:
It happens when the ESR endpoint is called, but in debugging it only throws the error on the second or subsequent call (/posts/{postId}/esr-reconciled) - let me try to find a slightly simpler use case to set up, because that's very clumsy!